src/GTP.cpp

```diff
@@ -515,18 +515,27 @@ bool GTP::execute(GameState & game, std::string xinput) {
     } else if (command.find("heatmap") == 0) {
         std::istringstream cmdstream(command);
         std::string tmp;
-        int rotation;
+        std::string rotation;
+        Network::Netresult vec;
 
         cmdstream >> tmp;   // eat heatmap
         cmdstream >> rotation;
 
-        if (!cmdstream.fail()) {
-            auto vec = Network::get_scored_moves(
-                &game, Network::Ensemble::DIRECT, rotation, true);
-            Network::show_heatmap(&game, vec, false);
+        if (!cmdstream.fail() && rotation == "all") {
+            for (int r=0; r<8; r++) {
+                vec = Network::get_scored_moves(
+                    &game, Network::Ensemble::DIRECT, r, true);
+                Network::show_heatmap(&game, vec, false);
+            }
         } else {
-            auto vec = Network::get_scored_moves(
-                &game, Network::Ensemble::DIRECT, 0, true);
+            if (!cmdstream.fail() && (rotation == "average" || rotation == "avg")) {
+                vec = Network::get_scored_moves(
+                    &game, Network::Ensemble::MULTI_AVG, 8, true);
+            } else {
+                vec = Network::get_scored_moves(
+                    &game, Network::Ensemble::DIRECT, cmdstream.fail() ? 0 : std::stoi(rotation), true);
```

Comment:
I'd move the ternary operator out of the function call for readability.

Reply:
Also this is really the only case that must handle cmdstream.fail(), so move that case out of this if and move it to the top? Then all further can assume that it did not fail.